promcda/models/ProMCDA.py

```diff
@@ -0,0 +1,519 @@
+        :return scores_df: pd.DataFrame or string
+        """
+        from promcda.utils import utils_for_parallelization
+        from promcda.models.mcda_without_robustness import MCDAWithoutRobustness
+        from promcda.configuration.configuration_validator import (check_indicator_weights_polarities,
+                                                                   process_indicators_and_weights)
+
+        # Configuration parameters' validation
+        if aggregation_method is not None:
+            if aggregation_method not in vars(AggregationFunctions).values():
+                raise ValueError(
+                    f"Invalid 'aggregation_method'. Expected one of {list(vars(AggregationFunctions).values())}, "
+                    f"got '{aggregation_method}'.")
+
+        index_column_name = self.input_matrix.index.name
+        index_column_values = self.input_matrix.index.tolist()
+
+        # Process indicators and weights based on input parameters in the configuration
+        input_matrix_no_alternatives, num_indicators, polarity, norm_weights = process_indicators_and_weights(self.input_matrix_no_alternatives,
+                                                        self.robustness_indicators,
+                                                        self.robustness_weights, self.robustness_single_weights,
+                                                        self.polarity, self.num_runs, self.weights, self.marginal_distributions)
+
+        # Check the number of indicators, weights, and polarities, assign random weights if uncertainty is enabled
+        try:
+            check_indicator_weights_polarities(num_indicators, polarity, robustness_weights=self.robustness_weights,
+                                               weights=norm_weights)
+        except ValueError as e:
+            logging.error(str(e), stack_info=True)
+            raise
+
+        # Apply aggregation in the different configuration settings
+        # NO UNCERTAINTY ON INDICATORS AND WEIGHTS
+        if not self.robustness_indicators and not self.robustness_weights and not self.robustness_single_weights:
+            mcda_without_robustness = MCDAWithoutRobustness(self.polarity, input_matrix_no_alternatives)
+            normalized_indicators = self.normalized_values_without_robustness
+            if normalized_indicators is None:
+                raise ValueError("Normalization must be performed before aggregation.")
+            if aggregation_method is None:
+                aggregated_scores = pd.DataFrame()
+                for agg_method in AggregationFunctions:
+                    result = mcda_without_robustness.aggregate_indicators(
+                        normalized_indicators=normalized_indicators,
+                        weights=norm_weights,
+                        agg_method=agg_method
+                    )
+                    aggregated_scores = pd.concat([aggregated_scores, result], axis=1)
```

Comment:
Here is the same case from mcda_without_robustness.py class. 
There is no advantage of concatenating result and aggregated_scores individually in if... else.

Reply:
As before, the reassignment to final_scores is intentional, as it accumulates the concatenated DataFrames progressively. This way, we keep adding new score data while preserving the previous ones. Without this, we would overwrite the existing data.